drv/transceivers-server/src/main.rs

```diff
@@ -715,6 +722,12 @@ fn main() -> ! {
             }
         }
 
+        server.in_a0 = match seq.tofino_seq_state() {
+            Ok(TofinoSeqState::A0) => true,
+            Ok(_) => false,
+            Err(_) => false,
+        };
```

Comment:
If I understand correctly, we only need to check the sequencer state on timer ticks that call `handle_i2c_loop` (which then calls `update_leds`). But, here we're performing the IPC to the sequencer task to get the power state on _every_ iteration of the loop, even on ones where we won't be messing with the LEDs.

What do you think about changing this so that, instead of having an `in_a0` field on `ServerImpl`, we instead passed the `TofinoSeqState` into `handle_i2c_loop` as an argument, which in turn passed it to `update_leds`? That way, we would only call the sequencer's `tofino_seq_state` IPC when we actually need to know what power state we're in, and the flow of that information to the place where it's actually consumed would be a bit more obvious.

Reply:
Your point about getting the IPC out of the main loop and into the slower loop is well taken! I've also reworked things to just pass the current sequencer state into the I2C loop as you suggest.